### PR TITLE
Add some missing typehints for `get_*` methods in client.py

### PR DESCRIPTION
**Pull request by MrKomodoDragon**
## Summary

This PR adds the missing typehints for the `id` parameter in some of the `get_*` methods for the `Client` class in client.py

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by NCPlayz — CHANGES_REQUESTED** [review]
These should all have positional only indicators too.

File: discord/client.py [hunk 1]
```diff
@@ -755,7 +755,7 @@ def get_partial_messageable(self, id: int, *, type: Optional[ChannelType] = None
         """
         return PartialMessageable(state=self._connection, id=id, type=type)
 
-    def get_stage_instance(self, id) -> Optional[StageInstance]:
+    def get_stage_instance(self, id: int) -> Optional[StageInstance]:
```
Comment (Rapptz) [inline discord/client.py:758]:
```suggestion
    def get_stage_instance(self, id: int, /) -> Optional[StageInstance]:
```

File: discord/client.py [hunk 2]
```diff
@@ -777,7 +777,7 @@ def get_stage_instance(self, id) -> Optional[StageInstance]:
         if isinstance(channel, StageChannel):
             return channel.instance
 
-    def get_guild(self, id) -> Optional[Guild]:
+    def get_guild(self, id: int) -> Optional[Guild]:
```
Comment (Rapptz) [inline discord/client.py:780]:
```suggestion
    def get_guild(self, id: int, /) -> Optional[Guild]:
```

File: discord/client.py [hunk 3]
```diff
@@ -792,7 +792,7 @@ def get_guild(self, id) -> Optional[Guild]:
         """
         return self._connection._get_guild(id)
 
-    def get_user(self, id) -> Optional[User]:
+    def get_user(self, id: int) -> Optional[User]:
```
Comment (Rapptz) [inline discord/client.py:795]:
```suggestion
    def get_user(self, id: int, /) -> Optional[User]:
```

File: discord/client.py [hunk 4]
```diff
@@ -807,7 +807,7 @@ def get_user(self, id) -> Optional[User]:
         """
         return self._connection.get_user(id)
 
-    def get_emoji(self, id) -> Optional[Emoji]:
+    def get_emoji(self, id: int) -> Optional[Emoji]:
```
Comment (Rapptz) [inline discord/client.py:810]:
```suggestion
    def get_emoji(self, id: int, /) -> Optional[Emoji]:
```

**Pull request merged**